articles/openshift/oshift-netpol.md

```diff
@@ -0,0 +1,138 @@
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: <desired_name_of_object>
+  namespace: <desired_project> (if not provided this will default to the project you are in when you run oc create)
+spec:
+```
+
+Inside the spec field we'll pass in the rules that will determine what our NetworkPolicy object does. The first thing you need to consider is the pods you want to target, this is done using the podSelector field. 
+You can use this to match all pods in a project or alternatively to match all pods with a certain label. If you leave this field blank it will match all pods. If you want to match pods with the label 'role=webserver' 
+it would look like:
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: webserver-egress 
+  namespace: webserver
+spec:
+  podSelector:
+    matchLabels:
+      role: webserver
+```
+
+At this point we've matched the pods inside our webserver project that have the label role=webserver. Let's say for example we want to allow these pods to connect to a service inside a project we've named 'database' we could
+achieve this using a namespace selector, which evaluates projects based on a label. In this case we are going to be looking for the namespace label database=true. Firstly we need to ensure this label is applied using the 
+following command:
+
+> [!NOTE]
+> The cluster administrator role is required to add labels to namespaces.
+
+``` oc label namespace database database=true ```
+
+Our netpol object will now look like:
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: webserver-egress
+  namespace: webserver
+spec:
+  podSelector:
+    matchLabels:
+      role: webserver
+  egress:
+  - to:
+    - namespaceSelector:
+        matchLabels:
+          database: "true"
+```
+
+We now have a template for a NetworkPolicy object that will match any pods labeled role=webserver inside the webserver project and allow them to communicate out to any projects labeled database=true. This is only half of the work,
+if we created this object now our desired connectivity would not work as we still need the ingress allowance inside the database project. The template for that object will look like:
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: database-ingress
+  namespace: database
+spec:
+  podSelector:
+    matchLabels:
+      role: database
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          webserver: "true"
+```
+
+This object will be created in the database project, it will match any pods in the project with the label "role=database" and allow ingress traffic from any project labeled "webserver=true" in order to achieve our desired outcome we will need to label our webserver project:
```

Comment:
```suggestion
This object will be created in the database project, it will match any pods in the project with the label "role=database" and allow ingress traffic from any project labeled "webserver=true". In order to achieve our desired outcome we will need to label our webserver project:
```